### PR TITLE
#9755 s2 s share requests directory

### LINKED ISSUE #9755: S2S_Revision of the 'share requests' directory_ Part 1 [3]
### Problem Description

 The 'share request' directory has no filter options. That is a problem, because after a while the directory becomes confusing.

At the moment the 'share requests' directory only shows incoming share requests and not the outgoing ones. That is a problem, because the health departments have no option to get an overview about shared cases/contacts and cannot track the shared cases/contacts.

Also the layout of the overview table does not include necessary information's, that the health departments need to be able to work.

### Proposed Change

- [ ] 1. Change the name from 'share requests' to 'share directory' (German: Übergabeverzeichnis)
- [ ] 2. Add the following filter options (layout and location are the same as in the case directory)

| Filter | Typ | Impact |
| ---| --- | --- | 
| Status | Dropdown | All - No records are filtered. <br/> Pending - Only shares with the value "Pending" in "Status" are displayed. <br/> Accepted - Only shares with the value "Accepted" in "Status" are displayed. <br/> Rejected - Only those shares are displayed that have the value "Rejected" in "Status". ~<br/> Error - Only shares with the value "Error" in "Status" are displayed.~<br/> "" is the default setting, which shows all entries | 

- [ ] 3. Delete the buttons 'Pending' and 'Accepted' 
image

- [ ] 4. Add the buttons 'Incoming' (German: Eingehend) in the place where the button 'Pending' was previously seen. 
image

| Button | Incoming | 
| :--- | :--- | 
| Impact | Only the incoming shares are displayed |
| Comment | -This filter is active by default when the share directory is opened for the first time in the session. <br/> -The total number of results for this filter are displayed on the button if the button has been clicked/ is active. <br/> -Clicking on the button 'Incoming' should show all cases/contacts that where shared from another health department to the own health department (status can be: accepted and pending- if user clicks 'reject' the share has to disappear because of data protection reasons). |

- [ ] 5. Add the button 'Outgoing' (German: Ausgehend) in the place where the button 'Accepted' was previously seen. 
image

| Button | Outgoing | 
| :--- | :--- | 
| Impact | Only the outgoing shares are displayed |
| Comment | -The total number of results for this filter are displayed on the button if the button has been clicked/ is active <br/> -Clicking on the button 'Outgoing' should show all cases/ contacts that where shared with another health department (status can be: accepted, pending, revoked and rejected) |

- [ ] 6. Change the configuration of the button 'All': This button should include all 'Incoming' cases/contacts and all 'Outgoing' cases/contacts. 
image

| Button | All | 
| :--- | :--- | 
| Impact | Both incoming and outgoing shares are displayed |
| Comment | The total number of results for this filter are displayed on the button if the button is clicked |

- [ ] 7. The layout of the overview table should include the following columns:

| | Share ID (German: Übergabe ID) | Share Date | Typ | From / To <br/> (Counterpart) | Sender Name | Ownership handed over | Status | Direction | | Comment |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| eye button <br/> A double click or a single click on the "eye" button opens the preview of a share (is already implemented) | First 6 sign of Share ID (already implemented) | The date of the share, localised according to current language (already implemented) | 'Case' or 'Contact' is shown (already implemented) | Name of other counterpart for example 'LK Würzburg' if 'LK Würzburg' is the sender organisation or received organisation | Name of Person who has initiated the share (Format: Last name, First name) | 'Yes' or 'No' is shown (already implemented) | Current status of the share 'accepted' 'pending' or 'rejected'. Is updated depending on what the target system decides (if the share is outgoing) or what the source system decides (if the share is incoming). | 'Incoming' or 'Outgoing'. | Accept and Reject button as they are already implemented (1.73.1) | Text field as already implemented (1.73.1) |

- [ ] 7.1 All columns can be sorted in ascending and descending order.

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #9755

### CODE REVIEW

**Review by JonasCir — CHANGES_REQUESTED** [review]
change_buttons
details_with
revoke_button
share_width

**Comment by leventegal-she** [comment]
> Please switch the buttons here. The reason is that users will more frequently select something and then apply it then revoke it. 

The order of the buttons are set in a base class so it is the same for all pages and I didn't see any complaints from SORMAS users to change it. So I would say it is ok.

> The address field is cut

You should scroll to right to see the address and other columns

> Please move the buttons at the end of the line after the comment. 

It is requested in the ticket that the comment is the last column

> The first column is cut although there is enough space

I will remove the caption of that column because it is not needed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/sormastosormas/ShareRequestGrid.java [hunk 1]
```diff
@@ -70,57 +76,64 @@ public ShareRequestGrid(boolean isInEagerMode, ShareRequestCriteria criteria) {
...
-		((Column<SormasToSormasShareRequestIndexDto, String>) getColumn(ExternalMessageIndexDto.UUID)).setRenderer(new UuidRenderer());
-		((Column<SormasToSormasShareRequestIndexDto, Date>) getColumn(SormasToSormasShareRequestIndexDto.CREATION_DATE))
+			ShareRequestIndexDto.UUID,
+			ShareRequestIndexDto.CREATION_DATE,
+			ShareRequestIndexDto.DATA_TYPE,
+			ShareRequestIndexDto.ORGANIZATION_NAME,
+			ShareRequestIndexDto.SENDER_NAME,
+			ShareRequestIndexDto.OWNERSHIP_HANDED_OVER,
+			ShareRequestIndexDto.STATUS,
+			COLUMN_ACTIONS,
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/sormastosormas/ShareRequestGrid.java:92]: It is requested in the ticket that the comment is the last column

**Comment by JonasCir** [comment]
@leventegal-she I'm not certain, are you still working on this? Are you going to address the inline comments from my last review? :)

**Comment by leventegal-she** [comment]
> you going to address the inline comments from my la

I changed as you  suggested

**Comment by JonasCir** [comment]
Ok I was confused as the comments were not resolved, my bad :) Good to go!

**Review by JonasCir — APPROVED** [review]

**Pull request merged**